duckduckgo/iOS
# Reattach orphaned bookmarks to root folder when reordering

Task/Issue URL: https://app.asana.com/0/0/1204720539208728/f

**Description**:
When moving a bookmark within root folder, consider orphaned bookmarks when validating to/from indexes and add
all affected orphans to the root folder.



**Steps to test this PR**:
1. Enable sync on 2 devices (another one can be macOS which simplifies testing) and have 2 folders:
* folder 1 with a number of bookmarks,
* folder 2 with at least 2 bookmarks.
3. Ensure that bookmarks on both devices are synced.
4. On iOS (the primary testing device) go offline and move some bookmarks from folder 1 to folder 2.
5. On macOS (or the secondary testing device) reorder the 2 bookmarks in folder 2.
6. Go online on iOS and trigger sync (i.e. by restarting the app) – your bookmarks move operations should be discarded since macOS reordering was newer and your moved bookmarks should end up in root folder as orphans.
7. Reorder bookmarks in the root folder by moving an orphan into non-orphans and vice versa. Verify that bookmarks are moved as expected.



**Copy Testing**:


**Orientation Testing**:


**Device Testing**:


**OS Testing**:


**Theme Testing**:


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
